src/nav-bar/NavBar.js

```diff
@@ -46,54 +52,39 @@ const addLinkClasses = ({ isCurrent }) => ({
   }),
 });
 
-export default function NavBar({ large, onClick }) {
+export default function NavBar({ pages, onClick, navigationStyles, nested }) {
   return (
     <NavContainer>
-      <NavList large={large}>
-        <NavItem>
-          <Link getProps={addLinkClasses} to={PATHS.overview} onClick={onClick}>
-            Overview
-          </Link>
-        </NavItem>
-        <NavItem>
-          <Link
-            getProps={addLinkClasses}
-            to={PATHS.sentencing}
-            onClick={onClick}
-          >
-            Sentencing
-          </Link>
-        </NavItem>
-        <NavItem>
-          <Link getProps={addLinkClasses} to={PATHS.prison} onClick={onClick}>
-            Prison
-          </Link>
-        </NavItem>
-        <NavItem>
-          <Link
-            getProps={addLinkClasses}
-            to={PATHS.probation}
-            onClick={onClick}
-          >
-            Probation
-          </Link>
-        </NavItem>
-        <NavItem>
-          <Link getProps={addLinkClasses} to={PATHS.parole} onClick={onClick}>
-            Parole
-          </Link>
-        </NavItem>
+      <NavList navigationStyles={navigationStyles}>
+        {Array.from(pages, ([path, label]) => (
+          <NavItem key={path}>
+            <Link
+              getProps={addLinkClasses}
+              to={`${nested ? "../" : ""}${PATHS[path]}`}
```

Comment:
This `nested` property is a bit hacky but without it, components that include the `NavBar` and are themselves children of the `Router`, will end up making invalid relative links.  For example, because the `OverviewPage` is nested under the main Router by default (i.e. without this `nested` property) it's generated navigation links would look like this:

`http://localhost:3000/us_nd/overview/parole` instead of the desired `http://localhost:3000/us_nd/parole`

Reply:
ew ... I can't come up with a better solution though so 👍